src/main/java/org/jenkinsci/maven/plugins/hpi/ValidateMojo.java

```diff
@@ -13,24 +13,16 @@
  */
 @Mojo(name = "validate", defaultPhase = LifecyclePhase.VALIDATE)
 public class ValidateMojo extends AbstractJenkinsMojo {
-    @Override
-    public void execute() throws MojoExecutionException, MojoFailureException {
-        if (!isMustangOrAbove())
-            throw new MojoExecutionException("JDK6 or later is necessary to build a Jenkins plugin");
 
-        if (new VersionNumber(findJenkinsVersion()).compareTo(new VersionNumber("1.419.99"))<=0)
-            throw new MojoExecutionException("This version of maven-hpi-plugin requires Jenkins 1.420 or later");
-    }
+    @Override
+    public void execute() throws MojoExecutionException {
+        JavaSpecificationVersion javaVersion = getMinimumJavaVersion();
+        if (JavaSpecificationVersion.forCurrentJVM().isOlderThan(javaVersion)) {
+            throw new MojoExecutionException("Java " + javaVersion + " or later is necessary to build this plugin.");
+        }
 
-    /**
-     * Are we running on JDK6 or above?
-     */
-    private static boolean isMustangOrAbove() {
-        try {
-            Class.forName("javax.annotation.processing.Processor");
-            return true;
-        } catch(ClassNotFoundException e) {
-            return false;
+        if (new VersionNumber(findJenkinsVersion()).compareTo(new VersionNumber("2.203.99")) <= 0) {
```

Comment:
Why not
```suggestion
        if (new VersionNumber(findJenkinsVersion()).compareTo(new VersionNumber("2.204")) < 0) {
```
?

Reply:
The same question could have been asked of the original code, but no particular reason. Trivial enough to fix while I am here I suppose.